services/api/src/resources/billing/resolvers.ts

```diff
@@ -0,0 +1,219 @@
+import * as R from 'ramda';
+import * as logger from '../../logger';
+import convertDateToMYSQLDateTimeFormat from '../../util/convertDateToMYSQLDateTimeFormat';
+import billingModel, { BillingModifier, getBillingModifier, BillingModifierBase } from '../../models/billing';
+import { BillingGroup, GroupInput } from '../../models/group';
+export interface BillingModifierInput extends BillingModifierBase {
+  group: GroupInput;
+}
+
+interface AddBillingModifierInput {
+  input: BillingModifierInput;
+}
+
+interface UpdateBillingModifierInput {
+  input: {
+    id: number;
+    patch: BillingModifier;
+  };
+}
+
+interface BillingModifiersInput {
+  input: GroupInput;
+  month?: string;
+}
+
+interface DeleteBillingModifierInput {
+  input: {
+    id: number;
+  };
+}
+
+interface DeleteAllBillingGroupModifiersInput {
+  input: GroupInput;
+}
+
+/**
+ * Create/Add Billing Modifier
+ *
+ * @param {obj} root The rootValue passed from the Apollo server configuration.
+ * @param {AddBillingModifierInput} args
+ * @param {ExpressContext} context this includes the context passed from the apolloServer query
+ *     { sqlClient, hasPermissions, keycloakGrant, requestCache }
+ *
+ * @return {BillingModifier} The created modifier
+ */
+type AddBillingModifierAlias = (
+  root: any,
+  args: AddBillingModifierInput,
+  context: { models: any; hasPermission: any }
+) => Promise<BillingModifier>;
+export const addBillingModifier: AddBillingModifierAlias = async (
+  _,
+  args,
+  context
+) => {
+  const { models, hasPermission } = context;
+  const { input } = args;
+
+  // Input Validation
+  if (R.isEmpty(input.group)) {
+    throw new Error('You must provide a billing group name or id');
+  }
+
+  if (R.isEmpty(R.omit(['group', 'startDate', 'endDate'], input))) {
+    throw new Error('You must provide a discount value or extra cost.');
+  }
+
+  if (new Date(input.startDate) >= new Date(input.endDate)) {
+    throw new Error('You must provide a start date before the end date.');
+  }
+
+  // Permissions
+  await hasPermission('group', 'add');
```

Comment:
We talked about how we probably need a whole new set of resources/scopes for billing modifiers. Just noting here for paper trail.